### PR TITLE
Add docs for intra-doc-links

### LINKED ISSUE #66000: Document intra-rustdoc links in the unstable book
https://github.com/rust-lang/rust/issues/43466

Also figure out where the docs belong when it stabilizes.

The implementation has diverged a bit from the original RFC (rust-lang/rfcs#1946). We should have some good docs on all the syntaxes supported.

cc @clarfon 

This issue has been assigned to @rcoh via [this comment](https://github.com/rust-lang/rust/issues/66000#issuecomment-605071840).

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/rust-lang/rust/issues/66000

### CODE REVIEW

**Review by jyn514 — CHANGES_REQUESTED** [review]
This should mention which scope is used to resolve the links: the parent module of the item. In the case of macros this is the crate root; in the case of re-exports it is the original module; in the case of impls I believe it's the parent of the type (but not sure).

**Comment by Manishearth** [comment]
@jyn514 isn't it just the def site scope? I don't think there's any special scoping here

**Comment by jyn514** [comment]
Sure, it's obvious to us because we implemented it, but it's not obvious to everyone else. See e.g. https://github.com/rust-lang/rust/issues/72243 which is a bug report that macros aren't resolved using lexical scope instead of resolution scope.

**Comment by Manishearth** [comment]
@jyn514 Oh, I do consider that special scoping, I wasn't aware of that. For everything else it's just the def site scope though. I think we can mention that `macro_rules` are scoped differently

**Comment by Manishearth** [comment]
Added a note. We'll be deprecating macro_rules anyway so I don't think we should do much more here

File: src/doc/rustdoc/src/intra-doc-links.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Intra-doc links
+
+_This feature is currently nightly-only_
+
+Rustdoc is capable of directly linking to other rustdoc pages in Markdown documentation using the path of item as a link.
+
+For example, in the following code all of the links will link to the rustdoc page for `Bar`:
+
+```rust
+
```
Comment (GuillaumeGomez) [inline src/doc/rustdoc/src/intra-doc-links.md:10]: Please remove this empty line.

**Comment by GuillaumeGomez** [comment]
Overall, it is nice. Could you also add the fact that using `crate::` or paths more generally work just like imports too please?

File: src/doc/rustdoc/src/intra-doc-links.md (hunk 1 above)
Comment (Manishearth) [inline src/doc/rustdoc/src/intra-doc-links.md:10]:
```suggestion
```

**Pull request closed without merging**